typing/typedecl.ml

```diff
@@ -2353,6 +2367,41 @@ let add_types_to_env decls shapes env =
       add_type ~check:true ~shape id decl env)
     decls shapes env
 
+let mark_decl_jkind_best decl =
+  match decl.type_kind with
+  | Type_abstract _ | Type_open -> decl
+  | Type_record (_, _) | Type_record_unboxed_product (_, _) | Type_variant (_, _) ->
+    { decl with type_jkind = Jkind.mark_best decl.type_jkind }
+
+(* Normalize the jkinds in a list of (potentially mutually recursive) type declarations *)
+let normalize_decl_jkinds env shapes decls =
+  (* Mark the jkinds in each decl as best (as we have now added all the information we
+     need to to mark them best) *)
+  let decls = List.map (fun (id, decl) -> (id, mark_decl_jkind_best decl)) decls in
```

Comment:
Is this step necessary? I think all the best markers should already be here, from `update_decls_jkind`.